test whitespace in pseudo-element selector throws syntax error

jacksonweekes opened the pull request:
<!-- Please describe your changes on the following line: -->
Issue #15448 - Land the test cases that should've landed with the selectors update. Tests that syntax error is thrown when pseudo-element selector contains whitespace - related to issue #15335 

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [X ] `./mach build -d` does not report any errors
- [X ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #15448  (github issue number if applicable).

<!-- Either: -->
- [ ] There are tests for these changes OR
- [X ] These changes do not require tests because implements a test not included when issue #15335 was fixed

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/16228)
<!-- Reviewable:end -->

jacksonweekes committed:
test whitespace in pseudo-element selector throws syntax error

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @jdm (or someone else) soon.

jacksonweekes committed:
fix formatting issue

emilio commented on tests/wpt/web-platform-tests/dom/nodes/selectors.js:
```diff
@@ -37,6 +37,8 @@ var invalidSelectors = [
   {name: "Unknown pseudo-element",       selector: "div::example"},
   {name: "Unknown pseudo-element",       selector: "::example"},
   {name: "Invalid pseudo-element",       selector: ":::before"},
+  {name: "Invalid pseudo-element",	     selector: ": before"},
```
nit: You have a tab in this line that should be converted to spaces.

emilio reviewed (commented):
Hi, thanks for doing this!

Seems like you need to update the test manifest so CI is green (`./mach update-manifest`).

We should also add tests for empty attribute selectors. There are some tests already, but we need to check `[attr$=""]` and `[attr=^=""]`.

Thanks again for looking into this!

jacksonweekes commented:
Hi @emilio thanks for reviewing! I have fixed the whitespace issue but just wanted to clarify about the checks for empty attribute selectors. I have added
`  {name: "Invalid [att=value] selector", selector: "[attr$=\"\"]"},
    {name: "Invalid [att=value] selector", selector: "[attr=^=\"\"]"},
`
to invalid selectors, however the first test fails. I'm not sure if I've misunderstood you or the parser is just not throwing an error for this case. Any advice?

emilio commented:
Sorry, those are not invalid selectors.

` [attr$=""]` is a valid selector, we should make sure it matches nothing. `[attr=^=""]` is indeed an invalid selector, but that's just because I made a typo (it should be `[attr^=""]`).

jdm commented:
@jacksonweekes Are you planning to finish this?

bors-servo commented:
:umbrella: The latest upstream changes (presumably #17822) made this pull request unmergeable. Please resolve the merge conflicts.

emilio commented:
Closing due to lack of activity.
